frontends/web/src/routes/device/bitbox02/createbackup.tsx

```diff
@@ -15,83 +15,70 @@
  * limitations under the License.
  */
 
-import { Component } from 'react';
-import { createBackup } from '../../../api/bitbox02';
-import { translate, TranslateProps } from '../../../decorators/translate';
+import { useState } from 'react';
+import { createBackup as createBackupAPI } from '../../../api/bitbox02';
 import { apiPost } from '../../../utils/request';
 import { alertUser } from '../../../components/alert/Alert';
 import { confirmation } from '../../../components/confirm/Confirm';
 import { Button } from '../../../components/forms';
 import { WaitDialog } from '../../../components/wait-dialog/wait-dialog';
+import { useTranslation } from 'react-i18next';
 
-interface CreateProps {
+type TProps = {
     deviceID: string;
 }
 
-type Props = CreateProps & TranslateProps;
+export const Create = ({ deviceID }: TProps) => {
+  const [creatingBackup, setCreatingBackup] = useState(false);
+  const [disabled, setDisabled] = useState(false);
+  const { t } = useTranslation();
 
-interface State {
-    creatingBackup: boolean;
-    disabled: boolean;
-}
-
-class Create extends Component<Props, State> {
-  public readonly state: State = {
-    creatingBackup: false,
-    disabled: false,
-  };
-
-  private createBackup = () => {
-    this.setState({ creatingBackup: true });
-    createBackup(this.props.deviceID)
-      .then(() => this.setState({ creatingBackup: false, disabled: false }))
+  const createBackup = () => {
+    setCreatingBackup(true);
+    createBackupAPI(deviceID)
+      .then(() => {
+        setCreatingBackup(false);
+        setDisabled(false);
+      })
       .catch(() => {
-        alertUser(this.props.t('backup.create.fail'));
+        setCreatingBackup(false);
+        setDisabled(false);
+        alertUser(t('backup.create.fail'));
       });
   };
 
-  private maybeCreateBackup = () => {
-    this.setState({ disabled: true });
-    apiPost('devices/bitbox02/' + this.props.deviceID + '/backups/check', {
+  const maybeCreateBackup = () => {
+    setDisabled(true);
+    apiPost(`devices/bitbox02/${deviceID}/backups/check`, {
```

Comment:
can you use `checkBackup` from `api/bitbox02`?

checkBackup promise is a bit different than the others, it resolves on success or rejects if there was no success.

```js

checkBackup(deviceId, true)
  .then((backupID) => console.log('success!', backupID))
  .catch(() => console.error('backup check not ok))
```

or with async await https://github.com/digitalbitbox/bitbox-wallet-app/blob/1fd8e8c63788ca9db46fa87b3a58618a699d8f4b/frontends/web/src/routes/device/bitbox02/checkbackup.tsx#L40-L41

using then/catch seemed nice at the time, but one problem is that errors in catch can't be typed, as there may be other unexpected errors. this is why I continued rewriting api calls without throwing if success is fails.

Reply:
Ah I see. Nice that we have `checkBackup`. Will do.